File: xpertmud/xpertmud/scripting/perl/helpConverter.pl
use strict;
use warnings;
use diagnostics;

use xmlparser;
use helplayout;

my $dtd = new helpdtd();
my $allHelp = new XMXmlNode('help');
$allHelp->setAttribute('name', 'help');

foreach my $file (@ARGV) {
  my $p = new XMXmlParser($dtd);
  $p->parseFile($file);
  foreach my $node (@{ $p->getNodes() }) {
    foreach my $inner (@{ $node->getSubNodes() }) {
      $allHelp->addChild($inner);
    }
  }
}

my $mode = "docbook";

my $layout = new helplayout($mode);
print $layout->generate($allHelp);

File: xpertmud/xpertmud/scripting/python/helpConverter.py
import sys
import re

from xmlparser import *
from helplayout import *

dtd = helpdtd()
allHelp = XMXmlNode('help')
allHelp.setAttribute('name', 'help')

for file in sys.argv:
    p = XMXmlParser(dtd)
    p.parseFile(file)
    for node in p.getNodes():
        for inner in node.getSubNodes():
            allHelp.addChild(inner)

mode = 'docbook'

if mode == "docbook":
    layout = helplayout('docbook')
    sys.stdout.write(layout.generate(allHelp))

